### DISCUSSION TITLE
Decimal value in primary_fungible_store::transfer_with_ref

### DISCUSSION THREAD

**Discussion by neethu-satheesh-s** [aptos-labs/aptos-developer-discussions#151] [Questions]
Hi, I have created a fungible digital asset 0x4 standard using primary_fungible_store::create_primary_store_enabled_fungible_asset with decimals as 8.  How do I provide the amount in octas in primary_fungible_store::transfer_with_ref(&transfer_ref, from_address, to, amount); The amount seems to be always taking the integer part. how to provide the decimal/offset values in octas? 

I'm expecting to give 100000000 in amount to represent asset worth 1  
& 150000000 in amount to represent asset worth 1.5 in
primary_fungible_store::transfer_with_ref(&transfer_ref, from_address, to, amount). Is this possible?

When I tested it was considering 100000000 amount as worth 100000000.
 Thanks.

**Comment by 0xjunha** [comment]
The decimal should be considered after getting value from the `balance` field. It is for display purpose.

If you have provided 100000000 as the `amount` argument with decimal of 8, it represents 1 coin, not 100000000 coins.

For example, if you have a Fungible Asset with symbol of "MOON" with decimal of 8, and you want to transfer 1.5MOON, the right value for the `amount` arg will be 150000000, not 1.5

**Reply by neethu-satheesh-s** [reply]
Thank you @0xjunha  .  I need to clarify one more thing that I'm using 0x4 standard using aptos_token_objects

The example of FA like Moon, works via coin::transfer<MOON> with amount 150000000 works to indicate 1.5MOON

How do we do the same for a 0x4 token standard created with  aptos_token_objects::token::create_named_token and primary_fungible_store::create_primary_store_enabled_fungible_asset
and transfer via with primary_fungible_store::transfer_with_ref(&transfer_ref, from_address, to, amount) ?

Thank you

**Reply by 0xjunha** [reply]
Works in the same way!
primary_fungible_store::transfer_with_ref(&transfer_ref, from_address, to, 150000000)
works for transferring 1.5MOON with 8 as decimal.

**Reply by neethu-satheesh-s** [reply]
Thank you @0xjunha  I did a test to try this, but it didnot work for me.

I gave the token limit as 100 and decimal.

I tried the amount as  primary_fungible_store::transfer_with_ref(&transfer_ref, from_address, to, 150000000)

It failed with the error:

assert!(store.balance >= amount, error::invalid_argument(EINSUFFICIENT_BALANCE));

error[E11001]: test failure
│     ┌─ /home/neethu/.move/https___github_com_aptos-labs_aptos-core_git_aptos-cli-v3.0.2/aptos-move/framework/aptos-framework/sources/fungible_asset.move:555:9
│     │
│ 549 │     fun withdraw_internal(
│     │         ----------------- In this function in 0x1::fungible_asset
│     ·
│ 555 │         assert!(store.balance >= amount, error::invalid_argument(EINSUFFICIENT_BALANCE));
│     │         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Test was not expected to error, but it aborted with code 65540 originating in the module 0000000000000000000000000000000000000000000000000000000000000001::fungible_asset rooted here
│ 


Did I miss anything? thanks

**Reply by neethu-satheesh-s** [reply]
I think I understand what I missed, I have to give the token limit in 10000000000 as well. Then it worked. Thanks @0xjunha  It was great help.

**Comment by neethu-satheesh-s** [comment]
Thank you @0xjunha  for helping me to resolve this.
